checks/evaluation/branch_protection_test.go

```diff
@@ -18,563 +18,1691 @@ import (
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "requiresApproversForPullRequests",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main":                      1,
+						"numberOfRequiredReviewers": 1,
+					},
+				},
+				{
+					Probe:   "requiresCodeOwnersReview",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "requiresLastPushApproval",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "requiresUpToDateBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "runsStatusChecksBeforeMerging",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "requiresPRsToChangeCode",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "blocksDeleteOnBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "blocksForcePushOnBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "branchesAreProtected",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "branchProtectionAppliesToAdmins",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "dismissesStaleReviews",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "requiresApproversForPullRequests",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2":                     1,
+						"numberOfRequiredReviewers": 1,
+					},
+				},
+				{
+					Probe:   "requiresCodeOwnersReview",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "requiresLastPushApproval",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "requiresUpToDateBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "runsStatusChecksBeforeMerging",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+				{
+					Probe:   "requiresPRsToChangeCode",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main2": 1,
+					},
+				},
+			},
+			result: scut.TestReturn{
+				Score:        8,
+				NumberOfWarn: 3,
+				NumberOfInfo: 7,
+			},
```

Comment:
if 1 branch is protected and 1 isn't, I would expect the score to be < 5?

Reply:
The `branchesAreProtected` probe covers these lines:

https://github.com/ossf/scorecard/blob/b03bd230e2e96cfc4ab3b4af1952d80a35865bc2/checks/evaluation/branch_protection.go#L73-L81

... which affect logging and not the score.

Reply:
Yes, but I think the findings here are an impossible combination of findings. If a branch isn't protected, the rest of the raw data would be empty.

Reply:
If you run it on an empty repo for example, you get the expected result: 
```console
$ go run main.go --repo https://github.com/spencerschrock/actions-test --checks Branch-Protection --show-details --format json --verbosity debug | jq
...
  "score": 0.0,
  "checks": [
    {
      "details": [
        "Warn: branch protection not enabled for branch 'main'"
      ],
      "score": 0,
      "reason": "branch protection not enabled on development/release branches",
      "name": "Branch-Protection",
...
```

Note: the probes are still there for all the branches. The probes are mainly `OutcomeNotAvailable` (two are `OutcomeNegative`), but because the branch isn't protected, it's not logged.

Reply:
a4896eb9988ae94ea4146ed902344462b0c59350